app/uk/gov/hmrc/eoricommoncomponent/frontend/services/SubscriptionDetailsService.scala

```diff
@@ -66,6 +66,30 @@ class SubscriptionDetailsService @Inject() (
       saveSubscriptionDetails(sd => sd.copy(contactDetails = Some(contactDetails)))
     }
 
+  def cacheContactAddressDetails(address: AddressViewModel, contactDetails: ContactDetailsModel)(implicit
+    hc: HeaderCarrier
+  ): Future[Unit] = {
+    val updatedAddress = address.copy(postcode = address.postcode.filter(_.nonEmpty))
```

Comment:
Why do we have this logic?

Reply:
this logic is needed to copy the address entry page values to the single contact details cache, we are merging the address details into the contact detail cache data.